phpBB/phpbb/auth/provider/db.php

```diff
@@ -151,22 +151,25 @@ public function login($username, $password)
 			$attempts = 0;
 		}
 
+		// Display a custom message for reCaptcha v3
+		$login_error_attempts = $this->config['captcha_plugin'] === 'core.captcha.plugins.recaptcha_v3' ? 'LOGIN_ERROR_ATTEMPTS_RECAPTCHA_V3' : 'LOGIN_ERROR_ATTEMPTS';
```

Comment:
I'm not sure if it wouldn't maybe make more sense to have this customizable per captcha plugin.

Reply:
It is not possible via code to know if you are using ReCaptcha v2 invisible but only Recaptcha v3. As for all others the usual message is fine.

Reply:
Proposals?

Reply:
I meant leave it up to the captcha implementation to define a custom message for login error attempts. I guess the captcha plugin should be able to determine itself if it's recaptcha v2 or v3?

Reply:
Well, not really. At least AFAIK for the reCatchaV2 thee are diffeent implementation that are not easily if not at all recognisable.
Unless I am missing something?

Reply:
The captcha plugin should know if it's `'core.captcha.plugins.recaptcha_v3'`, right? We already have the captcha factory in here. How about adding a method to `captcha_abstract` that adds a function `get_login_error_attempts()`? Afterwards you can use that inside the captcha plugins to return the language variable for this message. Therefore other plugins can use the same method as opposed to hardcoding this just for recaptcha v3.

Reply:
Recaptcha V2 has an invisible mode plus the normal standard, here it is where the logic fails due to the fact that the invisible one (V2) is not easily recognisable AFAIK.

Reply:
I agree this doesn't go here. The plugin should provide the key so any logic for determining the key should reside within the plugin. Otherwise what are you going to do when there is a v4 or a v5, etc.

Reply:
This are the vailable captchas
```
core.captcha.plugins.gd
core.captcha.plugins.gd_wave
core.captcha.plugins.nogd
core.captcha.plugins.qa
core.captcha.plugins.recaptcha
core.captcha.plugins.recaptcha_v3

```

Reply:
Did you plan on showing a different message for `core.captcha.plugins.recaptcha`? At least in this PR that does not seem to be intended. Why do you have to know this then? Even then, v2 could define a different error message with my proposal. Same goes for v3, gd, gd_wave, nogd, and qa. Or they could just display the default cause the abstract will default to the current message.

Reply:
> Did you plan on showing a different message for core.captcha.plugins.recaptcha? 

Nope... as I said:  The true intention of this PR is to give ReCaptcha-v3 a custom max login attempts message as per the title, that's what is doing ... instead of suppress it as per the ticket in the tracker.

> Why do you have to know this then?
Sorry? I knew that prior to push that PR... If I correctly understod.

> Please... let me see how or gimme some hint.

Reply:
@3D-I see this PR: https://github.com/phpbb/phpbb/pull/6155